### PR TITLE
feat: Add support to create template module README from template

### PR DESCRIPTION
**Pull request by akocbek**
### Description

Please include a summary of the change and which issue is fixed. Also, please include the motivation and context for the change. List any dependencies that are required for this change.

**Check the relevant boxes:**
- [ ] Bug fix (nonbreaking change that fixes an issue)
- [x] New feature (nonbreaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI related update (pipeline, etc.)

### Checklist

- [ ] If relevant, a test for the change has been added or updated as part of this PR.
- [ ] If relevant, documentation for the change has been added or updated as part of this PR.

### Merge

- Merge using "Squash and merge".
- Make sure to use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents. The commit message determines whether a new version of the modules needs to be released, and if so, which semver number to use).

### CODE REVIEW

File: module-assets/ci/terraformDocTemplateModule.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+def main():
+    # get repository name
+    my_command = "basename `git rev-parse --show-toplevel`"
+    proc = Popen(my_command, stdout=PIPE, stderr=PIPE, shell=True)
+    output, error = proc.communicate()
+    module_name = output.decode("utf-8").strip()
+
+    if proc.returncode != 0:
+        print(error)
+    elif module_name == "module-template":  # module-template
```
Comment (ocofaigh) [inline module-assets/ci/terraformDocTemplateModule.py:48]: what about [terraform-ibm-module-template](https://github.com/terraform-ibm-modules/terraform-ibm-module-template) ?

File: module-assets/.pre-commit-config.yaml [hunk 2]
```diff
@@ -112,6 +112,13 @@ repos:
       entry: ci/terraform-config-inspect.sh
       language: script
       pass_filenames: false
+    - id: build_module_template_readme
+      name: Build module template README
+      description: Build module template README (template_module only)
+      entry: python3 ci/terraformDocTemplateModule.py
+      language: python
+      files: "README.md"
+      pass_filenames: false
```
Comment (ocofaigh) [inline module-assets/.pre-commit-config.yaml:121]: Should the hook be moved above other hooks that may modify the readme? For example the "Add examples section to README" hook may have updates that need to apply after "Build module template README" hook runs

Comment (akocbek) [reply module-assets/.pre-commit-config.yaml:121]: At the moment we have hardcoded examples. if we change examples and move exampels hook after then we will have a deadlock where each of them will fix README file in a different way. (examples hook is updating the content inside module template hook)
to Update examples we will have to manually update hardcoded examples which need to be the same as the hook will create.

File: module-assets/ci/terraformDocTemplateModule.py (hunk 1 above)
Comment (akocbek) [reply module-assets/ci/terraformDocTemplateModule.py]: added

**Review by ocofaigh — COMMENTED** [review]
Is there a reason why the .yaml files are hidden? (start with a dot?)

**Review by ocofaigh — COMMENTED** [review]
maybe we move terraformDocTemplateModule.py into the new `module-template-automation` directory too to be consistent ?

**Comment by akocbek** [comment]
> Is there a reason why the .yaml files are hidden? (start with a dot?)

terraform_docs example is using hidden files for precommit-config yaml file https://github.com/terraform-docs/terraform-docs#pre-commit-hook. for consistency I used as well. But it is not needed.

**Review by ocofaigh — APPROVED** [review]

**Review by ocofaigh — APPROVED** [review]

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**